OPHD/UI/Core/RadioButton.cpp

```diff
@@ -33,60 +26,57 @@ RadioButton::~RadioButton()
  * \note	Setting check state with this method will not fire
  *			a clicked event.
  */
-void RadioButton::checked(bool toggle)
+void RadioButtonGroup::RadioButton::checked(bool toggle)
 {
 	mChecked = toggle;
 }
 
 
-bool RadioButton::checked() const
+bool RadioButtonGroup::RadioButton::checked() const
 {
 	return mChecked;
 }
 
-void RadioButton::parentContainer(UIContainer* parent)
+void RadioButtonGroup::RadioButton::parentContainer(RadioButtonGroup* parent)
 {
 	mParentContainer = parent;
 }
 
-void RadioButton::text(const std::string& text)
+void RadioButtonGroup::RadioButton::text(const std::string& text)
 {
 	mLabel.text(text);
 	onTextChange();
 }
 
-const std::string& RadioButton::text() const
+const std::string& RadioButtonGroup::RadioButton::text() const
 {
 	return mLabel.text();
 }
 
-RadioButton::ClickSignal::Source& RadioButton::click()
+
+RadioButtonGroup::RadioButton::ClickCallback& RadioButtonGroup::RadioButton::click()
 {
-	for (auto* sibling : mParentContainer->controls())
-	{
-		if (auto* asRadioButton = dynamic_cast<RadioButton*>(sibling))
-		{
-			asRadioButton->checked(false);
-		}
-	}
+	mParentContainer->clearSelection();
+	mRbgDelegate();
+
 	checked(true);
-	return mSignal;
+	return mCallback;
```

Comment:
Same, wrong method. This should go in the `onClick` handler method.

Reply:
ok, I don't have an `onClick()` method at this time, but I do have an `onMouseDown()` that now is used to execute the action.